src/components/ingestion/AbstractIngestionEngine.js

```diff
@@ -139,6 +114,33 @@ class AbstractIngestionEngine {
 		done(null, payload)
 	}
 
+	// The FFmpeg process used to wrap the ingestion stream in an MPEG-TS container
+	ffmpegProcess = null
```

Comment:
Very happy this all works now, but something feels wrong about having all of this stuff below instance methods.

For context: I had moved things out of the constructor for the sake of readability, but having things under methods makes me concerned they're a bit buried.

This is a genuine question: might it be better to have the nulled versions of these instances defined above the constructor, and then populate them in the constructor?

Does that even solve the problem, or do we face the same order challenges?

TL;DR -- I'm fine with leaving this as is and don't want to spend a million years moving it all around forever and ever, just wanted to raise it in case someone had a Strong Opinion

Reply:
Moved nulled properties above the constructor, set them in the constructor.